spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -696,4 +704,50 @@ private void cancelTaskExecution(TaskExecution taskExecution, String platformNam
 		return taskExecutions;
 	}
 
+	private void isCTRSplitValidForCurrentCTR(TaskLauncher taskLauncher, TaskDefinition taskDefinition) {
+		TaskParser taskParser = new TaskParser("composed-task-runner", taskDefinition.getProperties().get("graph"), true, true);
+		TaskNode taskNode = taskParser.parse();
+		ComposedRunnerVisitor composedRunnerVisitor = new ComposedRunnerVisitor();
+		taskNode.accept(composedRunnerVisitor);
+		if(composedRunnerVisitor.highCount > taskLauncher.getMaximumConcurrentTasks()) {
+			throw new IllegalArgumentException(String.format("One or more of the " +
+					"splits in the composed task contains a task count that exceeds " +
+					"the maximumConcurrentTasks count of %s",
+					taskLauncher.getMaximumConcurrentTasks()));
+		}
+	}
+
+
+	private class ComposedRunnerVisitor extends TaskVisitor {
```

Comment:
Is it possible to move this logic into `spring-cloud-dataflow-core-dsl`?

Reply:
That is a good question.  I don't think `spring-cloud-dataflow-core-dsl` is a good home for this logic.    This module contains a parser and as thus as a parser it provides syntax checking and the ability for users to utilize its visitor callbacks.    The parser doesn't have knowledge of the rest of SCDF or any restrictions that SCDF applies to itself, i.e.  maximum number of concurrrent tasks.   Also this check  is not a syntax check, its more of a runtime check.  
Thoughts?

Reply:
I agree this is more of a runtime check but the way I see is that the `ComposedRunnerVisitor` acts just as an extension of the actual `TaskVisitor` with the `high` and `current` counts which only carries the abstract counts on the split and after all this moves into the data flow DSL anyway ;) 

Also, I meant to move just this `ComposedRunnerVisitor` not the other logic associated with it.
Perhaps we could rename the ComposedRunnerVisitor to ComposedTaskRunnerVisitor as well.

Reply:
but but but... I thought I won that argument :-D
Sounds good !

Reply:
The game is still on ;)  do you think any other impact on moving this into `spring-cloud-dataflow-core-dsl` ?

Reply:
Updated the PR based on your recommendation.  It was a good one!

Reply:
Thank you @cppwfs. Looks good to me. Merging.